snippets/c_api/iterate_events.cc

```diff
@@ -0,0 +1,197 @@
+/*******************************************************************************
+ The MIT License (MIT)
+
+ Copyright (c) 2016 Maps.Me
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ *******************************************************************************/
+
+// This define is needed to preserve client's timestamps in events.
+#define ALOHALYTICS_SERVER
+#define EXPORT extern "C"
+
+#include <algorithm>
+#include <iostream>
+#include <set>
+#include <string>
+
+#include "../../src/event_base.h"
+#include "../../src/processor.h"
+
+// Public struct used in Python.
+struct UserInfo
+{
+  char os;
+  float lat;
+  float lon;
+  char raw_uid[32];
+
+  UserInfo()
+  : os(0), lat(0), lon(0)
+  {
+    std::memset(raw_uid, 0, 32);
```

Comment:
BTW, please, create a named constant for 32 - I saw it several times in the code.